src/main/java/org/folio/circulation/domain/representations/changeduedate/BatchChangeDueDateRequest.java

```diff
@@ -0,0 +1,76 @@
+package org.folio.circulation.domain.representations.changeduedate;
+
+import static org.folio.circulation.support.JsonPropertyFetcher.getDateTimeProperty;
+import static org.folio.circulation.support.Result.succeeded;
+import static org.folio.circulation.support.ValidationErrorFailure.failedValidation;
+
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+import org.folio.circulation.support.Result;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BatchChangeDueDateRequest {
+
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  private static final String LOAN_IDS = "loanIds";
+  private static final String DUE_DATE = "dueDate";
+
+  private final List<String> loanIds;
+  private final DateTime dueDate;
+
+  private BatchChangeDueDateRequest(List<String> loanIds,
+    DateTime dueDate) {
+
+    this.loanIds = loanIds;
+    this.dueDate = dueDate;
+  }
+
+  public static Result<BatchChangeDueDateRequest> from(RoutingContext routingContext) {
+
+    List<String> loanIds;
+    DateTime dueDate;
+    try {
+      JsonObject payload = routingContext.getBodyAsJson();
+      if (payload == null) {
+        return failedValidation("No payload provided",
```

Comment:
Should it be bad request instead maybe?

Reply:
Maybe it should. 422 is commonly used in FOLIO, so it was used in this case. What benefits does 400 bring?

Reply:
No benefits, I just think that 400 code better describes the failure. 

As far as I understand the `422` means  a validation issue, so a field(s) in the request has a not valid value/has no value.

But in this case the request has not supplied at all, it sounds to me more like bad request rather then a validation failure. Also, as far as I remember RMB returns 400 in such cases, but it worth to test it.

Does it make sense?

Reply:
It makes litytle sence because everywhere in mod-circulation 422 is returned in case of validation errors.

Reply:
@bohdan-suprun is correct in his description. There is a defacto convention that requests that cannot be parsed are responded to with a 400 rather than a 422 with validation errors.

The motivation for this could be that [a description](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422) of 422 is a follows:
`The HyperText Transfer Protocol (HTTP) 422 Unprocessable Entity response status code indicates that the server understands the content type of the request entity, and the syntax of the request entity is correct, but it was unable to process the contained instructions.`

Which suggests that is only appropriate for situations where the server has understood the request, which in this case it hasn't.